File: backend/models/order.go
package models

import "gorm.io/gorm"

type Order struct {
	gorm.Model
	UserID     uint        `json:"user_id"`
	User       User        `json:"user"`
	TotalPrice float64     `json:"total_price"`
	Status     string      `json:"status"` // e.g., pending, paid, shipped
	OrderItems []OrderItem `json:"items"`
	Payment    Payment     `json:"payment"`
	AddressID  uint        `json:"address_id"`
	Address    Address     `json:"address"`
}

type OrderItem struct {
	gorm.Model
	OrderID   uint    `json:"order_id"`
	ProductID uint    `json:"product_id"`
	Product   Product `json:"product"`
	Quantity  int     `json:"quantity"`
	Price     float64 `json:"price"` // price at time of purchase
}

File: frontend/src/types/Order.ts
import type { OrderItem } from "./OrderItem";
import type { Address } from "./Address";

export interface Order {
  ID: number;
  user_id: number;
  status: string;
  total_price: number;
  address_id: number;
  address?: Address;
  items: OrderItem[];
  created_at: string;
  updated_at: string;
  
}
